clang/lib/Sema/SemaExpr.cpp

```diff
@@ -5813,6 +5813,23 @@ static TypoCorrection TryTypoCorrectionForCall(Sema &S, Expr *Fn,
   return TypoCorrection();
 }
 
+static bool isParenthetizedAndQualifiedAddressOfExpr(Expr *Fn) {
+  if (!isa<ParenExpr>(Fn))
+    return false;
+
+  Fn = Fn->IgnoreParens();
```

Comment:
Should we also ignore implicit casts along the way here for all of these?

Reply:
Nope, Any cast would make it a different shape than `(&foo::bar)`